ccm_web/server/src/no.cache.interceptor.ts

```diff
@@ -0,0 +1,15 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor } from '@nestjs/common'
+import { Observable } from 'rxjs'
+
+import { Response as ExpressResponse } from 'express'
+
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching#the_cache-control_header
+
+@Injectable()
+export class CacheControlToHeaderInterceptor implements NestInterceptor {
+  intercept (context: ExecutionContext, next: CallHandler): Observable<unknown> {
+    const ResponseObj: ExpressResponse = context.switchToHttp().getResponse()
+    ResponseObj.setHeader('Cache-Control', 'no-store')
```

Comment:
I don't know too much about what headers are appropriate here either.

I'm looking at Canvas and they just set
`max-age=0`

But they're also using this if-none-match and etag to improve performance. I'd be a little concerned about the performance of just setting it to `no-store` without considering etags. I don't know how much effort that would take in a nestjs app though. Maybe this is related? https://wanago.io/2022/01/17/api-nestjs-etag-cache/

Reply:
> max-age=0

I just noticed it my self just now. I think this will bad. I am going to change it back not in the `main.ts` and put in the `api.controller.ts` and see if that fixes that.

Reply:
I tested again on canvas beta server for Cache-control header for the static file `main.ts`. I do see randomly it introduces max-age =0. So I was thinking it might be my code doing it, but probably not. Express/NestJS handling it some how. Chatting with @jonespm said Due to Etag this might be controlled and *might* not be an issue. But need to talk to Ross about it

Reply:
@ssciolla and @jonespm , Ross said he will test and let us know if we **really** need to add cache headers or default express Etag step is good enough.

Reply:
After our discussion in today's meeting. I did testing in 4 scenarios looking for Status code response time for a browser fetch (GET) for static assets like (`main.js`, I only looked at this) and an API response (like  get sections)

Scenarios tested:
1. Code change from this branch/PR, Local run, Configured on Canvas Dev
2. Code Change mixing changes from this PR and Sam's PR #309, Local run, Configured on Canvas Dev
3. CCM dev instance configured on Canvas test, running latest from git `Main`
4. CCM Test instance configured on Canvas Beta, running latest from git `Main`

My finding are recored [here](https://docs.google.com/spreadsheets/d/1Wu085vyFmuDBuVpYaxmz6vKazQoonAa1oevy3ZEVicE/edit#gid=0)

@ssciolla and @jonespm Let me know your thoughts. My conclusion is adding this change won't be disruptive.

Reply:
Okay sounds like there will be no issue with what you are doing. I will do some local testing this afternoon and then approve if I don't see any issues.

Reply:
@ssciolla I want you to pull this changes in your #316 and test it.